templates/components/agents/python/contract_review/app/ui/states/workflow.py

```diff
@@ -0,0 +1,133 @@
+from typing import Any, Dict, List, Optional
+
+import reflex as rx
+from pydantic import BaseModel
+
+from app.services.contract_reviewer import LogEvent
+from rxconfig import config as rx_config
+
+
+class ContractLoaderState(rx.State):
+    is_running: bool = False
+    is_started: bool = False
+    log: List[Dict[str, Any]] = []
+
+    @rx.event
+    async def add_log(self, log: LogEvent):
+        if not self.is_started:
+            yield ContractLoaderState.start
+        self.log.append(log.model_dump())
+        if "data saved" in log.msg:
+            self.is_running = False
+
+    def has_log(self):
+        return len(self.log) > 0
+
+    @rx.event
+    def start(self):
+        self.is_running = True
+        self.is_started = True
+
+    @rx.event
+    def stop(self):
+        self.is_running = False
+
+    @rx.event
+    def reset_state(self):
+        self.is_running = False
+        self.is_started = False
+        self.log = []
+
+
+class GuidelineState(rx.State):
+    is_running: bool = False
+    is_started: bool = False
+    log: List[Dict[str, Any]] = []
+
+    @rx.event
+    def add_log(self, log: LogEvent):
+        if not self.is_started:
+            yield GuidelineState.start
+        self.log.append(log.model_dump())
+        if "Dispatched" in log.msg:
+            yield GuidelineState.stop
+
+    def has_log(self):
+        return len(self.log) > 0
+
+    @rx.event
+    def reset_state(self):
+        self.is_running = False
+        self.is_started = False
+        self.log = []
+
+    @rx.event
+    def start(self):
+        self.is_running = True
+        self.is_started = True
+
+    @rx.event
+    def stop(self):
+        self.is_running = False
+
+
+class GuidelineData(BaseModel):
+    is_completed: bool
+    is_compliant: Optional[bool]
+    clause_text: Optional[str]
+    output: Optional[Dict[str, Any]]
+
+
+class GuidelineHandlerState(rx.State):
```

Comment:
As the guideline step has a nested data but Reflex is really hard to handle state with complex data so i split it into two models: GuidelineState and GuidelineHandlerState